backend/src/main/java/com/bakdata/conquery/apiv1/QueryProcessor.java

```diff
@@ -400,42 +436,56 @@ public ExternalUploadResult uploadEntities(Subject subject, Dataset dataset, Ext
 	 */
 	public FullExecutionStatus getSingleEntityExport(Subject subject, UriBuilder uriBuilder, String idKind, String entity, List<Connector> sources, Dataset dataset, Range<LocalDate> dateRange) {
 
-		final ConceptQuery entitySelectQuery =
-				new ConceptQuery(new CQDateRestriction(Objects.requireNonNullElse(dateRange, Range.all()), new CQExternal(List.of(idKind), new String[][]{{"HEAD"}, {entity}}, false)));
+		final ConceptQuery entitySelectQuery = new ConceptQuery(new CQExternal(List.of(idKind), new String[][]{{"HEAD"}, {entity}}, true));
 
 		final TableExportQuery exportQuery = new TableExportQuery(entitySelectQuery);
-		exportQuery.setTables(
-				sources.stream()
-					   .map(source -> {
-						   final CQConcept cqConcept = new CQConcept();
-						   cqConcept.setElements(List.of(source.getConcept()));
-						   final CQTable cqTable = new CQTable();
-						   cqTable.setConcept(cqConcept);
-						   cqTable.setConnector(source);
-						   cqConcept.setTables(List.of(cqTable));
+		exportQuery.setDateRange(dateRange);
+
+		exportQuery.setTables(sources.stream().map(CQConcept::forConnector).collect(Collectors.toList()));
 
-						   return cqConcept;
-					   })
-					   .collect(Collectors.toList())
-		);
+		final ManagedQuery execution = (ManagedQuery) postQuery(dataset, exportQuery, subject, true);
 
-		final ManagedExecution<?> execution = postQuery(dataset, exportQuery, subject, true);
+		final List<Select> infoCardSelects = getInfoCardSelects(config.getPreview(), dataset, datasetRegistry);
+
+
+		final AbsoluteFormQuery infoCardQuery =
+				new AbsoluteFormQuery(entitySelectQuery, dateRange,
+									  ArrayConceptQuery.createFromFeatures(
+											  infoCardSelects.stream()
+															 .map(CQConcept::forSelect)
+															 .collect(Collectors.toList())),
+									  List.of(ExportForm.ResolutionAndAlignment.of(Resolution.COMPLETE, Alignment.NO_ALIGN))
+				);
 
+		final ManagedQuery infoCardExecution = (ManagedQuery) postQuery(dataset, infoCardQuery, subject, true);
 
-		// collect id immediately so it does not get sucked into closure
-		final ManagedExecutionId id = execution.getId();
 
-		while (execution.awaitDone(10, TimeUnit.SECONDS) == ExecutionState.RUNNING) {
-			log.trace("Still waiting for {}", id);
+		if (execution.awaitDone(10, TimeUnit.SECONDS) == ExecutionState.RUNNING) {
+			log.warn("Still waiting for {} after 10Seconds.", execution.getId());
```

Comment:
10 ist schon ganz schön knapp wenn es gerade fast 9 sec braucht, oder?
```suggestion
			log.warn("Still waiting for {} after 10 seconds.", execution.getId());
```

Reply:
> 10 ist schon ganz schön knapp wenn es gerade fast 9 sec braucht, oder?

Was meinst du mit "wenn es gerade fast 9 sec braucht"?